src/Client/ClientBase.cpp

```diff
@@ -1979,9 +1979,8 @@ void ClientBase::readArguments(
         }
         /// Options with value after equal sign.
         else if (in_external_group
-            && (0 == strncmp(arg, "--file=", strlen("--file=")) || 0 == strncmp(arg, "--name=", strlen("--name="))
-                || 0 == strncmp(arg, "--format=", strlen("--format=")) || 0 == strncmp(arg, "--structure=", strlen("--structure="))
-                || 0 == strncmp(arg, "--types=", strlen("--types="))))
+            && (arg == "--file="sv || arg == "--name="sv || arg == "--format="sv
+                || arg == "--structure="sv || arg == "--types="sv))
```

Comment:
This wont work. For example when we have `--file=file_name` the value of `arg` will be `"--file=file_name`" and we will compare it with `"--file="` and it will fail

Reply:
Oh, so it's a `startsWith` semantic.

Reply:
But why do we do exact match for variant without equal sign?

Reply:
I think we can use `starts_with` from C++20 here to simplify the code